### PR TITLE
(closes #216) Tag section in VM and template details

### LINKED ISSUE #216: Implement tags tab for VM and images (templates)
Tags pane must be located when other tabs are located. It must support CRUD operations for tags, every tag is key-value, where key/value is 255 chars max length field. (Task for copying templates - #239

Tags must be lexically sorted and grouped by first part (the part is the tag name field string before first dot), e.g. "vm.".

```
VM
  vm.name1 = ...
  vm.name2 = ...
```

```
Credentials
  credentials.login = ...
  credentials.password = ...
```

If tag doesn't include "." then all those tags must be groupped into "Common" tab.

The frontend must support searching by substring with search field. 

This task is prerequisite for #2

### PR DESCRIPTION
**Pull request by wowshakhov**

### CODE REVIEW

**Comment by bwsw** [comment]
@wowshakhov
@zolotyx 
@ershovaelena 

It's completely buggy.
New category addition doesn't work right, it doesn't redraw the pane. Edit icon is positioned badly. If the section includes "+", why I see the dropdown to choose it. If you use two line Key/Value placement, then mark it K/V or somekind, but I think two line display is a bad way. May be use stair-like placement at least:

```
Key
    Value
```

If the PR is not for inclusion, let it go, otherwise, It's not ready for merge at all.

**Comment by wowshakhov** [comment]
@bwsw There was a problem with Jenkins build, so you probably saw an outdated version. The latest version should be up now.

**Comment by bwsw** [comment]
@zolotyx @wowshakhov @ershovaelena 
looks great now

File: src/app/tags/tags.component.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  private getSearchResults(): Array<TagCategory> {
+    let categories = cloneDeep(this.categories);
+
+    if (!this.query) {
+      return categories;
+    }
+
+    categories = categories.map(category => {
+      category.tags = this.filterTags(category.tags);
+      return category.tags.length ? category : false;
```
Comment (zolotyx) [inline src/app/tags/tags.component.ts]: get rid of ternary operator

File: src/app/tags/tag-category/tag-category.component.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Taggable } from '../../shared/interfaces/taggable.interface';
+import { Tag } from '../../shared/models';
```
Comment (zolotyx) [inline src/app/tags/tag-category/tag-category.component.ts]: let's directly include tag model

File: src/app/tags/tags-view/tags-view.component.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    this.onTagDelete.emit(tag);
+  }
+
+  public updateResults(): void {
+    this.categories = this.getCategories();
+    this.updateSearchResults();
+  }
+
+  public updateSearchResults(): void {
+    this.visibleCategories = this.getSearchResults();
```
Comment (zolotyx) [inline src/app/tags/tags-view/tags-view.component.ts]: Please make it accept query string

File: src/app/tags/tags-view/tags-view.component.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  private getSearchResults(): Array<TagCategory> {
+    let categories = cloneDeep(this.categories);
+
+    if (!this.query) {
+      return categories;
+    }
+
+    categories = categories.map(category => {
```
Comment (zolotyx) [inline src/app/tags/tags-view/tags-view.component.ts]: Remove condition from map (change filter)

File: src/app/template/template-tags/template-tags.component.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<cs-tags-view
+  [tags]="tags.getValue()"
```
Comment (zolotyx) [inline src/app/template/template-tags/template-tags.component.html]: use [tags]="tags$ | async"
(rename tags to tags$) to show that it's Observable

**Pull request merged**